pkg/userdata/flatcar/provider.go

```diff
@@ -254,66 +256,15 @@ systemd:
         WantedBy=multi-user.target
 
     - name: kubelet.service
-      enabled: true
-      contents: |
-        [Unit]
-        Description=Kubernetes Kubelet
-        Requires=docker.service
-        After=docker.service
-        [Service]
-        TimeoutStartSec=5min
-        CPUAccounting=true
-        MemoryAccounting=true
-        EnvironmentFile=-/etc/environment
-        EnvironmentFile=/etc/kubernetes/nodeip.conf
```

Comment:
we need this file

Reply:
this is done via the kubeletFlags macro.
You can see it from the expected config here https://github.com/kubermatic/machine-controller/pull/1067/files/a173b5a6aedbf5fbec5461c77ac6077f563db568#diff-1f679fb68a00fca3c52634cc0620d2dc5ee938e7e821bc6790f1233b8acf082cL99-R101

Reply:
kubeletFlags uses the ENV that should be set by `EnvironmentFile=/etc/kubernetes/nodeip.conf`, but since it's removed, there is no `$KUBELET_NODE_IP` ENV.

Reply:
@kron4eg got your point, will check tomorrow why it was left behind, thx for the finding